### ISSUE TITLE
Failed to install on Mac OS X 10.8.5

### ISSUE THREAD

**Issue by fingolfin** [wbhart/Nemo.jl#12]
I tried installing Nemo during your presentation in Hamburg, but it run into an error while install flint. This is on Mac OS X 10.8.5, thus a fairly old system (but I have my reasons for sticking with that old system -- in particular, "never change a running system"; but I might cave in once Mac OS X 10.11 is out... *sigh*)

Next to mention, I am using julia 0.5.0-dev+321 (commit b939592) -- perhaps that is "too new", since you explicitly mention Julia 0.4, but it seems somewhat unlikely to cause the build error below. But if you think it might be cause, I can retry with 0.4rc2 or so.

Anyway, I next did
```
Pkg.clone("https://github.com/wbhart/Nemo.jl")
Pkg.build("Nemo")
```

It started to build a bunch of stuff, including e.g. GNU m4, etc. At this point I should point out that I am using [Fink](http://www.finkproject.org/), and thus have e.g. GNU m4, various clang and GCC versions, etc., all installed into the /sw prefix -- but I am not sure how to tell Julia resp. Nemo to use stuff from that (although I do have global CPPFLAGS and LDFLAGS set with the appropriate -I and -L flags, though it may not pick up on these). That said, I don't mind if it prefers to build its own, but I mention this as it might be cause of the error (e.g. because different GMP / MPIR versions get mixed up)

The error then finally looks like this:
```
    CC   ../build/fq_zech_poly_factor/pow.lo
    CC   ../build/fq_zech_poly_factor/print.lo
    CC   ../build/fq_zech_poly_factor/factor_kaltofen_shoup.lo
    CC   ../build/fq_zech_poly_factor/factor_squarefree.lo
    CC   ../build/fq_zech_poly_factor/factor_cantor_zassenhaus.lo
    CC   ../build/fq_zech_poly_factor/is_squarefree.lo
    CC   ../build/fq_zech_poly_factor/../fq_zech_poly_factor.lo
make[2]: Leaving directory '/Users/mhorn/.julia/v0.5/Nemo/deps/flint2/fq_zech_poly_factor'
clang: warning: argument unused during compilation: '-ansi'
Undefined symbols for architecture x86_64:
  "___gmpf_fits_si_p", referenced from:
      _fmpz_set_mpf in fmpz.lo
  "___gmpn_com_n", referenced from:
      _fmpz_bit_unpack in fmpz.lo
      _fmpz_bit_pack in fmpz.lo
  "___gmpn_neg_n", referenced from:
      __fmpz_vec_get_fft in fmpz_vec.lo
      __fmpz_vec_set_fft in fmpz_vec.lo
      _flint_mpn_mulmod_2expp1_basecase in mpn_extras.lo
      _fft_butterfly_sqrt2 in fft.lo
      _fft_adjust_sqrt2 in fft.lo
      _fft_butterfly_twiddle in fft.lo
      _ifft_negacyclic in fft.lo
      ...
  "___gmpn_sumdiff_n", referenced from:
      _butterfly_lshB in fft.lo
      _butterfly_rshB in fft.lo
  "___gmpn_tdiv_q", referenced from:
      _flint_mpn_tdiv_q_fmpz_inplace in fmpz_poly.lo
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:142: recipe for target 'libflint-13.5.2.dylib' failed
make[1]: *** [libflint-13.5.2.dylib] Error 1
make[1]: Leaving directory '/Users/mhorn/.julia/v0.5/Nemo/deps/flint2'
Makefile:173: recipe for target 'library' failed
make: *** [library] Error 2
===============================================================[ ERROR: Nemo ]================================================================

LoadError: failed process: Process(`make -j4`, ProcessExited(2)) [2]
while loading /Users/mhorn/.julia/v0.5/Nemo/deps/build.jl, in expression starting on line 132

==============================================================================================================================================
```

**Comment by wbhart** [comment]
Hi Max,

Thanks for trying to get it working.

As the Julia website says, you can try your luck with nightlies, but it may
not work. I can't say anything about it as they change every day, but the
last 0.5 nightly was broken for me. This is not unexpected.

Having said that, this is not causing your issue.

Julia doesn't provide gcc or the build environment. That is provided by
your system. Julia only provides a shell basically.

We also don't use any system provided libraries, but build them ourselves.

What I don't understand about the error message is that it appears to not
be finding the gmp library after it is built (or more precisely the symbols
in it), despite the fact that we tell the linker precisely where to find
it.

Is it somehow possible that it has built a 32 bit GMP or something?

Take a look in the file ~/.julia/v0.4/Nemo/deps/build.jl

and thinking of it as a simple shell script, does this look something like
the set of commands you would issue to build those dependencies on your
system?

Basically if you can build all those things on your system it should be
possible from Julia.

Having said that, we have another OSX user who has Xcode installed and who
gets an error saying the compiler can't build a binary when trying to
configure the first dependency, even though they can ordinarily build
software on their system.

I knew Windows was difficult, but I had no idea Apple had so massively
complicated building software on their systems.

Bill.

**Comment by wbhart** [comment]
Apparently the m4 build system (autotools) causes issues on OSX.

But after that, everything builds except Pari. So the issues you are having
I am guessing are due to your environment.

My best guess is GMP/MPIR is getting the symbol prefix wrong.

I think the best you can do is to try and build each dependency by hand
yourself first and try to write a simple C program against it to ensure it
works. Probably one of our dependencies just doesn't build correctly on
your system.

Maybe by building them by hand outside of the Julia shell we can narrow
down what needs to be patched.

Bill.

**Comment by wbhart** [comment]
We just issued 0.3.1 which fixes the pari issue and the m4 issue.

I don't have any insight into the symbol issue you reported, but it is not
a problem with a  recent OSX toolchain. We have one user on OSX now without
issues.

Bill.
